src/components/App/App.js

```diff
@@ -22,6 +23,18 @@ type Props = {
 };
 
 class App extends Component<Props> {
+  componentDidMount() {
+    window.addEventListener('beforeunload', this.killAllRunningProcesses);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('beforeunload', this.killAllRunningProcesses);
+  }
+
+  killAllRunningProcesses = () => {
+    ipcRenderer.send('killAllRunningProcesses');
+  };
```

Comment:
I'm curious if you have a way to test this? It looks like now that we're forwarding electron logs to the client, they don't actually appear (since the page immediately unloads).

It does look like it works :) just curious if there's a way to see logs I'm missing

Reply:
It should be relatively trivial, `ipcRenderer` is a component of electron so it logs to the shell running Electron, not the browser console. If you log from within `main.js`, it should persist in the shell through browser reloads.

Reply:
I think the issue is that we're launching the electron process from within the create-react-app start script, and then forwarding any Electron logs _to_ the browser window (at least, I think that's what we're doing).

Because yeah in the terminal, I just see webpack output.

We probably should find a way around that 🤔 although I guess it only matters in these rare cases